.github/workflows/deploy.yml

```diff
@@ -0,0 +1,30 @@
+name: Deploy
```

Comment:
Can we rename it to something more meaningful? This workflow is not only about deploying. Perhaps something like `pr-ci`. Also could you update the file name accordingly?